agent/connect/ca/provider_vault.go

```diff
@@ -709,6 +722,66 @@ func (v *VaultProvider) Stop() {
 
 func (v *VaultProvider) PrimaryUsesIntermediate() {}
 
+// We use raw path here
+func (v *VaultProvider) mountNamespaced(namespace, path string, mountInfo *vaultapi.MountInput) error {
+	defer v.client.SetNamespace(v.baseNamespace)
+	if namespace != "" {
+		v.client.SetNamespace(namespace)
+	}
```

Comment:
Do we need to guard the provider with a mutex to prevent a race condition here?

Reply:
I'd convinced myself that this couldn't happen. But seems safer to have the mutex, so added it.